provider/core/src/resource.rs

```diff
@@ -172,15 +246,67 @@ impl ResourceKey {
     /// use icu_provider::prelude::*;
     ///
     /// let resc_key = icu_provider::hello_world::key::HELLO_WORLD_V1;
-    /// let components = resc_key.get_components();
+    /// let components: Vec<&str> = resc_key
+    ///     .iter_components()
+    ///     .collect();
     ///
     /// assert_eq!(
     ///     ["core", "helloworld@1"],
-    ///     components.iter().collect::<Vec<&str>>()[..]
+    ///     components[..]
     /// );
     /// ```
-    pub fn get_components(&self) -> ResourceKeyComponents {
-        self.into()
+    pub fn iter_components(&self) -> impl Iterator<Item = &str> {
```

Comment:
Now that we have `get_path`, I don't see why we still (publicly) need this. Before this was the cheapest way to return the path, but now we can just return the static ref. So instead of doing `path_buf.extend(key.iter_components())` we can do `path_buf.push(key.get_path())`.

Reply:
#1516